### DISCUSSION TITLE
How to store secrets?

### DISCUSSION THREAD

**Discussion by s4m-mo** [deta/space-docs#66] [Q&A] [answered]
How can I store secrets (but not in the Spacefile) I want to share my Spacefile with my team and commit to a GitHub repo. The Spacefile contains the secrets as environment variables, but when I share those secrets, they will be public knowledge. How can I push environment variables and secrets to my Deta Space Micro without committing it anywhere in my code?

Thanks for your time.

**Comment by abdelhai** [comment, marked as answer]
you're not supposed to set the value in the Spacefile, but rather in the UI of the app.

https://user-images.githubusercontent.com/1752577/227964240-a774f7e7-153e-4c9e-8570-6bd8a03ce34b.mp4

**Reply by s4m-mo** [reply]
Thank you

**Reply by lete114** [reply]
> you're not supposed to set the value in the Spacefile, but rather in the UI of the app.

Why is there no `Configuration` option after I open `Settings`?

**Reply by s4m-mo** [reply]
> > you're not supposed to set the value in the Spacefile, but rather in the UI of the app.
> 
> Why is there no `Configuration` option after I open `Settings`?

Have you opened your account settings or the settings for your Deta App Project?

**Reply by lete114** [reply]
![image](https://github.com/deta/space-docs/assets/48512251/66be6639-a310-40e3-995f-ec502e5b40ec)
![image](https://github.com/deta/space-docs/assets/48512251/53a1e599-7719-4100-bda1-840248c81a93)

**Reply by s4m-mo** [reply]
Hmmm. That baffles me.

**Reply by lete114** [reply]
I recently read the documentation carefully and found that you need to set `presets.env` to have the `Configuration` option

https://deta.space/docs/en/basics/micros#custom-variables
